website/content/docs/discovery/dns.mdx

```diff
@@ -96,6 +96,41 @@ pairs according to [RFC1464](https://www.ietf.org/rfc/rfc1464.txt).
 Alternatively, the TXT record will only include the node's metadata value when the
 node's metadata key starts with `rfc1035-`.
 
+## Alternative Domain
+
+By default, Consul responds to DNS queries only for its configured
+[`domain`](/docs/agent/options#domain).
+
+Some use cases require responding to queries for more than one domain,
+such as during a DNS migration or to distinguish between internal and
+external queries by using different domains.
+
+Consul can be configured to respond to DNS queries on an alternative domain
+through the [`alt_domain`](/docs/agent/options#alt_domain) agent configuration
+option. Consul's DNS response will use the same domain as was used in the query.
+For example, if `test-domain` is configured as the alternative domain, the following query:
```

Comment:
```suggestion
By default, Consul responds to DNS queries in the `consul` domain, but you can set a specific domain for responding to DNS queries by configuring the [`domain`](/docs/agent/options#domain) parameter.

You can specify an alternative domain for DNS queries in the `alt_domain` parameter. Configuring the [`alt_domain`](/docs/agent/options#alt_domain) agent parameter enables Consul to respond to DNS queries to the specified domain.

In the following example, the `alt_domain` parameter is set to `test-domain`:

<CodeTabs heading="agent.hcl">
<CodeBlockConfig>

```hcl
alt_domain = "test-domain"
```

</CodeBlockConfig>
<CodeBlockConfig>

```json
"alt_domain" : "test-domain"
```

</CodeBlockConfig>
</CodeTabs>

The configuration enables the following `test-domain` query to run successfully:
```
I was just trying out some alternate wording to see if I could make it smoother. Please ignore if it doesn't seem like an improvement. 
I also wanted to show an example of the `alt_domain` configuration. This is because it wasn't clear from looking at the entry in the reference whether the option took a Boolean value to enable alt domains or if you have to specify the actual domain as a string.

Reply:
My personal opinion is that the new wording leaves out some important information (e.g., why someone might want to use an alternative domain, what domain is used in the response). I personally prefer the previous wording.

However, it is problematic if the `alt_domain` configuration explanation doesn't make its use clear.

@trujillo-adam : Does it make more sense to update the description on the agent configuration option page to make its use more obvious (and perhaps include an example)? Or perhaps provide the configuration example here as you suggested, but also link to this "Alternative Domain" section on the DNS page from the `alt_domain` agent config option description?

Reply:
Ah, that's because I accidentally left a paragraph out that I planned on putting back in at line 102:

"In some instances, Consul may need to respond to queries in more than one domain, such as during a DNS migration or to distinguish between internal and external queries. "

As far as updating the alt_domain reference, yes, we should update it so include at least the data type and default. As far as cross-linking, we should probably think about it a little more. We ca either put 0 guidance in the reference and link internal pages for usage or vice versa. We're currently a little inconsistent. For now I think just adding a little more detail to the reference section is good enough.

Reply:
@kbabuadze - can you add the HCL configuration example @trujillo-adam suggested?